packages/cli/src/commands/run-all-tests/run-all-tests.command.ts

```diff
@@ -99,55 +118,34 @@ const handler: (options: Options) => Promise<void> = async ({
         config,
         client,
         testRun,
+        executionOptions,
+        screenshottingOptions,
         apiToken,
         commitSha,
         appUrl,
         useAssetsSnapshottedInBaseSimulation,
-        headless,
-        devTools,
-        bypassCSP,
-        diffThreshold,
-        diffPixelThreshold,
-        padTime,
-        shiftTime,
-        networkStubbing,
-        parallelTasks,
+        parallelTasks: parallelTasks ?? undefined,
```

Comment:
The fact that numbers can come through as null is an implementation detail of the fact we use yargs, so I tried to isolate this as early as possible, and not let the nulls propagate too far through our codebase (just use undefineds instead). Note that optional params, absent keys etc. translate into undefined, so if we want to only use one of null vs undefined, it's normally easier to just use undefined (this is the reason we standardised on undefined at Palantir, and I think also [at Microsoft](https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined)).

Reply:
> STOP READING IMMEDIATELY
> THIS PAGE PROBABLY DOES NOT PERTAIN TO YOU.
> 
> These are Coding Guidelines for Contributors to TypeScript. This is NOT a prescriptive guideline for the TypeScript community. These guidelines are meant for contributors to the TypeScript project's codebase. We have chosen many of them for team consistency. Feel free to adopt them for your own team.
> 
> AGAIN: This is NOT a prescriptive guideline for the TypeScript community

I previously tried `undefined` everywhere but it ended up reading worse in many circumstances

Reply:
Yep those are TS's internal guidelines, but I think they use most of them more widely across Microsoft. I've found them useful, because although the tradeoffs for TypeScript vs other codebases can be different, the TS team has thought about them carefully.

> I previously tried undefined everywhere but it ended up reading worse in many circumstances

Do you have an example? Is this re. using undefined vs null? Are you proposing using null instead of undefined, or using empty string instead of null or undefined?

Can just chat when we catch up if that's easier.